client/core/wallet.go

```diff
@@ -105,7 +105,7 @@ func (w *xcWallet) state() *WalletState {
 	return &WalletState{
 		Symbol:       unbip(w.AssetID),
 		AssetID:      w.AssetID,
-		Open:         w.locallyUnlocked(),
+		Open:         len(w.encPW) == 0 || len(w.pw) > 0,
```

Comment:
The recursive RLock shouldn't be a problem though, right?  I haven't made any effort to diagnose https://github.com/decred/dcrdex/pull/850/checks?check_run_id=1443539618#step:5:406 however.

Reply:
I think it can, yes. The other place the mutex is hanging is in a call to `setBalance`, which is a write-lock. If the order is rlock1 -> lock -> rlock2, it'll hang. You'd have to sneak that lock in there just right, but an audit of the places the `xcWallet.mtx` is locked indicated that this was the only possible place something could happen.

Reply:
Ah, that's right if the lock is right between the two rlocks.  Good catch.